### ISSUE TITLE
Source nushell script only if the conditions met

### ISSUE THREAD

**Issue by atahabaki** [nushell/nushell#7988]
### Describe the bug

My whole $nu.config-path file is:

```nu
if $nu.os-info.name == "windows" {
    source ~\AppData\Roaming\nushell\starship.nu
} else if $nu.os-info.name == "linux" {
    source ~/.config/nushell/starship.nu
}
```

I suppose, the nushell is trying to find file while in parsing mode?
In windows:

```
Error: nu::parser::parser_incomplete (link)

  × Parser incomplete.
   ╭─[C:\Users\atahabaki\AppData\Roaming\nushell\config.nu:2:1]
 2 │     source ~\AppData\Roaming\nushell\starship.nu
 3 │ } else if $nu.os-info.name == "linux" {
   ·        ─┬
   ·         ╰── parser support missing for this expression
 4 │     source ~/.config/nushell/starship.nu
   ╰────
```

In linux:

```
Error: nu::parser::sourced_file_not_found (link)

  × File not found
   ╭─[/home/atahabaki/.config/nushell/config.nu:1:1]
 1 │ if $nu.os-info.name == "windows" {
 2 │     source ~\AppData\Roaming\nushell\starship.nu
   ·            ──────────────────┬──────────────────
   ·                              ╰── File not found: ~\AppData\Roaming\nushell\starship.nu
 3 │ } else if $nu.os-info.name == "linux" {
   ╰────
  help: sourced files need to be available before your script is run
```

### How to reproduce

1. Place `source file.nu` inside a conditional if statement,
2. Whoila, does not work, even tries to find the file whether conditions met or not.

### Expected behavior

Source the given file after the conditions met.

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.75.0                                   |
| branch             |                                          |
| commit_hash        | 17a265b197ad6bb77db5a07942a9f160ec260eb0 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| pkg_version        | 0.75.0                                   |
| build_time         | 2023-02-01 08:01:00 -06:00               |
| build_rust_channel | release                                  |
| features           | database, default, trash, which, zip     |
| installed_plugins  |                                          |

**Comment by atahabaki** [comment]
> Another common issue is trying to dynamically create the filename to source from:

from [Thinking in Nu](https://www.nushell.sh/book/thinking_in_nu.html#think-of-nushell-as-a-compiled-language)

I got the point, but the parser (In my opinion) should not try to find the file. I'm not loading it, it's loading after the conditions met.

I don't say this should be fixed. What I'm trying to say is completely different. This approach kills the conditional flow completely.

**Comment by fdncred** [comment]
One thing you can play around with is using `const`, because you can set const variables and still source them. We may have a bug though in our if/else if expressions.
```
const winpath = '~\AppData\Roaming\nushell\starship.nu'
const linpath = '~/.config/nushell/starship.nu'
```
This seems to work for me 
```
if $nu.os-info.name == "windows" {
    source $winpath
}
```
but I can't get it to work with and `else if` and I'm not sure why, which is why i say, we may have a bug.

Let me know if you make any progress with this.

**Comment by Mehrbod2002** [comment]
@fdncred 
Can I work on it ?

**Comment by fdncred** [comment]
@Mehrbod2002 You can source `const` variables from within `if` loops but there's currently a bug preventing the if/else if part.

**Comment by atahabaki** [comment]
I was creating another nushell script and I find out that nushell can't get the else statement if it's placed after if's right bracket, to demonstrate:
```nu
if $mylocalvar == "sth1" {
  do sth1
}
else if $mylocalvar == "sth2" {
  do sth2
}
```

Errors, that _nushell_ don't know why the else is there or something similar. Moving `else if` to same line as the previous line, if's right bracket, I mean:

```nu
if $mylocalvar == "sth1" {
  do sth1
} else if $mylocalvar == "sth2" {
  do sth2
}
```

No warning, and works correct. Weird, probably needs an another issue...

**Comment by fdncred** [comment]
ya, i've seen things like that before. agreed, another issue would be good if there's not one already.

**Comment by atahabaki** [comment]
@1Kinoti thx! That was helpful.

**Comment by 1Kinoti** [comment]
i believe this is [fixed](https://www.nushell.sh/blog/2023-09-19-nushell_0_85_0.html#improvements-to-parse-time-evaluation) now

**Comment by fdncred** [comment]
Agreed. With const additions, you should be able to do things similar to the OP, but with minor changes.
